src/uct/cuda/cuda_copy/cuda_copy_md.c

```diff
@@ -368,6 +397,25 @@ uct_cuda_copy_md_query_attributes(uct_cuda_copy_md_t *md, const void *address,
 
     mem_info->type = UCS_MEMORY_TYPE_CUDA;
 
+    /* Synchronize future DMA operations */
+#if HAVE_CUDA_FABRIC
+    if (is_vmm_allocation) {
+        cu_err = cuCtxSetFlags(CU_CTX_SYNC_MEMOPS);
```

Comment:
I think that the setting this flag is unexpected for the user.

Reply:
It is required for VMM memory in case this memory gets passed to IB transports after a cu*mempcy operation in the application.

Reply:
Maybe it is better to check flags here, and print an error to ask user to set the flag. Because this will change synchronization behavior for any operation on this context... what do you think?

Reply:
Previously synchronization behavior was forced onto the specific memory region that was passed but now it applies to all VMM allocations created under the device context. It's the scope that changes mostly. Returning an error would be a pain point for applications that use libraries that allocate VMM memory under the hood. IMO it's ok to enforce synchronization behavior on the context.